src/datatype.jl

```diff
@@ -0,0 +1,36 @@
+# custom data tyoe to hold ImageHDU
+struct CCDData{T,M<:AbstractMatrix{T}} <: AbstractMatrix{T}
+    data::M
+    hdr::FITSHeader
+end
+
+# extending the AbstractMatrix interface
+Base.size(A::CCDData) = size(A.data)
+Base.getindex(A::CCDData, i) = getindex(A.data, i)
+Base.setindex!(A::CCDData, v, i) = setindex!(A.data, v, i)
+Base.eachindex(A::CCDData) = eachindex(A.data)
+Base.iterate(A::CCDData) = iterate(A.data, state=(eachindex(A),))
+Base.length(A::CCDData) = length(A.data)
+Base.axes(A::CCDData) = axes(A.data)
+Base.IndexStyle(::CCDData) = IndexCartesian()
+
+read_hdu(hdu::ImageHDU) = CCDData(getdata(hdu), read_header(hdu))
```

Comment:
I think this function name should just be `CCDData` since it's acting as a constructor
```suggestion
CCDData(hdu::ImageHDU) = CCDData(getdata(hdu), read_header(hdu))
```